### ISSUE TITLE
(SnapDeals) once a sector is marked for snap, new deals will be added to it even the sector is inactive

### ISSUE THREAD

**Issue by Reiers** [filecoin-project/lotus#8011]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [X] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.14.0-rc5+butterflynet+git.40485eb88+api1.5.0
Local: lotus version 1.14.0-rc5+butterflynet+git.40485eb88
```

### Describe the Bug

If you sectors snap-up -> miss wdPost -> get a deal -> that deal will go inside the non-active sector -> and you are left with:

(SEE LOG)

on loop. This has potential to bring the cluster down.
Say you have a batch of 200 deals in span of 24h, you have upgraded the sectors. But something wrong happen with wdPost. Sector become unactive, but they are still receiving deals. So it will just pile on and on, unless you abort every single sector Upgrade.
**Left unattended** - you have a cluster of workers with the sector/deals files repeating this over and over again.
Long story short - it will be bad.

**Solution:** if wdPost fails - all sectors in that partition should `abortUpgrade` go back to Proving instantly. 
_So you dont end up with full workers and a confused scheduler._

### Logging Information

```text
2022-01-29T22:26:49.020+0100	ERROR	sectors	storage-sealing/states_replica_update.go:154	handleSubmitReplicaUpdate: error sending message: GasEstimateMessageGas error: estimating gas used: message execution failed: exit 16, reason: no valid updates (RetCode=16)
2022-01-29T22:26:49.025+0100	INFO	sectors	storage-sealing/states_failed.go:28	ReplicaUpdateFailed(29), waiting 59.97434873s before retrying
```

### Repo Steps

- `lotus-miner sectors snap-up <sectornumb>`
- miss wdPost 
- get a deal
- The deal will go inside the non-active sector

**Comment by arajasek** [comment]
There's a few aspects to this:

- we should not add deals to inactive sectors (perhaps based on startepoch)
- through the pipeline if we detect a sector is inactive we should...do something?
  - the correct answer is probably to finish any proof computation, then "pause" -- we can try submitting the message in 12 hours or so?
  - the simpler answer may just be to abort -- or add UX to indicate which sectors are good abort candidates if resources are constrained
  - an abort-all command for snapdeals might be useful too

**Comment by arajasek** [comment]
The scenario we are concerned about is:

- 200 sectors are marked to be snapped up
- they have deals added and are in the computation stage
- they go inactive due to a missed post
- they are now unproveable for the next 24 hours (though the correct proof can still be computed), but will keep retrying / hogging resources

The underlying problem is that the SubmitMessage stage shouldn't use any valuable resources if a correct proof has already been computed, and doesn't need to be done by a commit worker.

We also probably wanna use the finalizeEarly step that CommitAggregate uses to make sure valid replicas can move to long term storage

**Comment by arajasek** [comment]
Let's say we'll at least do:

> we should not add deals to inactive sectors (perhaps based on startepoch)

for 1.14.0

**Comment by arajasek** [comment]
#7905 might be relevant

**Comment by jennijuju** [comment]
@ZenGround0 suggested we might be over complicating things - he will prototype a PR to demonstrate the fix later and we can continue discussion there

**Comment by jennijuju** [comment]
closed by #8014
